imageio/core/util.py

```diff
@@ -46,39 +47,69 @@ def urlopen(*args, **kwargs):
     return urlopen(*args, **kwargs)
 
 
-def image_as_uint8(im):
-    """ Convert the given image to uint8
-    
-    If the dtype is already uint8, it is returned as-is. If the image
-    is float, and all values are between 0 and 1, the values are
-    multiplied by 255. In all other situations, the values are scaled
-    such that the minimum value becomes 0 and the maximum value becomes
-    255.
+def image_as_uint(img_in, bitdepth=None):
+    """ Convert the given image to uint (default: uint8)
+    
+    If the dtype already matches the desired format, it is returned
+    as-is. If the image is float, and all values are between 0 and 1,
+    the values are multiplied by np.power(2.0, bitdepth). In all other
+    situations, the values are scaled such that the minimum value
+    becomes 0 and the maximum value becomes np.power(2.0, bitdepth)-1
+    (255 for 8-bit and 65535 for 16-bit).
     """
-    if not isinstance(im, np.ndarray):
-        raise ValueError('image must be a numpy array')
+    if not bitdepth:
+        bitdepth = 8
+    if not isinstance(img_in, np.ndarray):
+        raise ValueError('Image must be a numpy array')
+    if bitdepth == 8:
+        out_type = np.uint8
+    elif bitdepth == 16:
+        out_type = np.uint16
+    else:
+        raise ValueError('Bitdepth must be either 8 or 16')
+    im = img_in.copy()
     dtype_str = str(im.dtype)
-    # Already uint8?
-    if dtype_str == 'uint8':
+    if ((im.dtype == np.uint8 and bitdepth == 8) or
+       (im.dtype == np.uint16 and bitdepth == 16)):
+        # Already the correct format? Return as-is
         return im
-    # Handle float
-    mi, ma = np.nanmin(im), np.nanmax(im)
-    if dtype_str.startswith('float'):
-        if mi >= 0 and ma <= 1:
-            mi, ma = 0, 1
-    # Now make float copy before we scale
-    im = im.astype('float32')
```

Comment:
This line makes a copy of im (see [numpy.ndarray.astype](http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.ndarray.astype.html)). When it was removed, I needed to make sure that the input image was not changed in place.